app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationHelper.java

```diff
@@ -317,7 +317,8 @@ public static void updateSummaryNotifications(@NonNull Context context, @NonNull
             // Create a notification that summarises the other notifications in this group
 
             // All notifications in this group have the same type, so get it from the first.
-            Notification.Type notificationType = (Notification.Type)members.get(0).getNotification().extras.getSerializable(EXTRA_NOTIFICATION_TYPE);
+            String typeName = members.get(0).getNotification().extras.getString(EXTRA_NOTIFICATION_TYPE, Notification.Type.UNKNOWN.name());
+            Notification.Type notificationType = Notification.Type.byString(typeName);
```

Comment:
since `MainActivity.openNotificationIntent` only needs the type name, we could skip this and pass the name as String directly?

Reply:
I'd say the "Type" is the proper information level for `openNotificationIntent`. That we internally use a string for the extra I would characterize as workaround, so I wouldn't put that in the "interface".

Reply:
Ok but be careful. `byString` is not how to recreate the enum from `name`. `valueOf` is.